build.gradle

```diff
@@ -294,3 +294,41 @@ release {
     pushToRemote = ''
   }
 }
+
+// Reports linkage errors across multiple Apache Beam artifact ids.
+//
+// To use (from the root of project):
+//    ./gradlew -Ppublishing -PjavaLinkageArtifacts=artifactId1,artifactId2,... :checkJavaLinkage
+//
+// For example:
+//    ./gradlew -Ppublishing -PjavaLinkageArtifacts=beam-sdks-java-core,beam-sdks-java-io-jdbc :checkJavaLinkage
+//
+// Note that this task publishes artifacts into your local Maven repository.
+if (project.hasProperty('javaLinkageArtifactIds')) {
+  if (!project.hasProperty('publishing')) {
+    throw new GradleException('You can only check linkage of Java artifacts if you specify -Ppublishing on the command line as well.')
+  }
+
+  configurations { linkageCheckerJava }
+  dependencies {
+    linkageCheckerJava "com.google.cloud.tools:dependencies:1.0.1"
+  }
+
+  // We need to evaluate all the projects first so that we can find depend on all the
+  // publishMavenJavaPublicationToMavenLocal tasks below.
+  for (p in rootProject.subprojects) {
+    if (!p.path.equals(project.path)) {
+      evaluationDependsOn(p.path)
+    }
+  }
+
+  project.task('checkJavaLinkage', type: JavaExec) {
+    dependsOn project.getTasksByName('publishMavenJavaPublicationToMavenLocal', true /* recursively */)
```

Comment:
Nice. I was wondering how to bridge Gradle build to Maven local repo. Good to know this task.

Reply:
Can we publish somewhere under `build/` instead, or to a tempdir, so that we don't have to install to the user's `.m2` cache?

Reply:
Linkage Checker does not support other directory than `~/.m2` as of now. This design follows Maven's decision not to allow users to set maven repository directory through M2_HOME environmental variable (since Maven 3.5.0 https://maven.apache.org/docs/3.5.0/release-notes.html)

Reply:
I tried using the `publishMavenJavaPublicationTotestPublicationRepository` since that is configured to go to `root_of_project/testPublication` but task for some reason also published to Maven local making using it moot.

I could not figure out an alternative which did not mess with the users Maven local.

Reply:
OK. Well it is a snapshot anyhow and better to have linkage checking in a funky way than not have it. Let's file improvements against LinkageChecker